esp-hal-common/src/usb_serial_jtag.rs

```diff
@@ -344,36 +500,36 @@ impl embedded_io::Write for UsbSerialJtag<'_> {
 
 #[cfg(feature = "async")]
 mod asynch {
-    use core::task::Poll;
+    use core::{marker::PhantomData, task::Poll};
 
     use embassy_sync::waitqueue::AtomicWaker;
     use procmacros::interrupt;
 
-    use super::{Error, Instance};
-    use crate::UsbSerialJtag;
+    use super::{Error, UsbSerialJtag, UsbSerialJtagRx, UsbSerialJtagTx};
+    use crate::peripherals::USB_DEVICE;
 
     // Single static instance of the waker
     static WAKER: AtomicWaker = AtomicWaker::new();
```

Comment:
We now need two wakers, one for RX and one for TX. Previously, because it was one struct you could only read or write, now we can do both at the same time meaning we need to be able to register a waker for each.

Reply:
Ahh oops, thanks for pointing that out. Should be fixed now!